### PR TITLE
Fix for updating product variant without providing an attributes

### PR DESCRIPTION
**Pull request by GrzegorzDerdak**
This PR allows updating a product variant without providing attributes.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/product/mutations/products.py [hunk 1]
```diff
@@ -1300,14 +1295,20 @@ def clean_input(
...
-                    attributes, product_type
-                )
+                if attributes:
+                    cls.validate_duplicated_attribute_values(
+                        attributes, used_attribute_values, instance
+                    )
+                    cleaned_input["attributes"] = cls.clean_attributes(
+                        attributes, product_type
+                    )
+                elif not instance.pk and not attributes:
```
Comment (IKarbowiak) [inline saleor/graphql/product/mutations/products.py:1307]: Please add a comment here. It's hard to understand what happened at first sight.

**Review by maarcingebala — COMMENTED** [review]
@GrzegorzDerdak Do we have tests for this particular case that this PR is fixing?

**Review by fowczarek — APPROVED** [review]

**Comment by GrzegorzDerdak** [comment]
> @GrzegorzDerdak Do we have tests for this particular case that this PR is fixing?

Yes we do, that is why I removed the attributes from test cases to make sure these changes are tested properly.

**Comment by maarcingebala** [comment]
@GrzegorzDerdak The tests are failing.

**Comment by GrzegorzDerdak** [comment]
Locally everything passes, I guess it's related to Travis CI, can I re-run it somehow?

**Comment by maarcingebala** [comment]
I've restarted them and now it's good :)

**Pull request merged**